src/at.rs

```diff
@@ -119,122 +119,122 @@ impl<Ctx, T: At<Ctx>> At<Ctx> for Option<&T> {
-		assert_eq!(val.at("hello").at(0), Some(&Value::u128(1)));
-		// References to valid locations are fine too:
-		assert_eq!(val.at(&&"hello").at(&&&0), Some(&Value::u128(1)));
-	}
-
-	#[test]
-	fn accessing_variants() {
-		let val = Value::named_variant(
-			"TheVariant",
-			[("foo", Value::u128(12345)), ("bar", Value::char('c'))],
-		);
-
-		assert_eq!(val.at("foo").unwrap().as_u128().unwrap(), 12345);
-		assert_eq!(val.at("bar").unwrap().as_char().unwrap(), 'c');
-
-		let val = Value::unnamed_variant("TheVariant", [Value::u128(12345), Value::char('c')]);
-
-		assert_eq!(val.at(0).unwrap().as_u128().unwrap(), 12345);
-		assert_eq!(val.at(1).unwrap().as_char().unwrap(), 'c');
-
-		// We can use `at()` on the variant directly, too:
-
-		let val = Variant::named_fields(
-			"TheVariant",
-			[("foo", Value::u128(12345)), ("bar", Value::char('c'))],
-		);
-
-		assert_eq!(val.at("foo").unwrap().as_u128().unwrap(), 12345);
-		assert_eq!(val.at("bar").unwrap().as_char().unwrap(), 'c');
-
-		let val = Variant::unnamed_fields("TheVariant", [Value::u128(12345), Value::char('c')]);
-
-		assert_eq!(val.at(0).unwrap().as_u128().unwrap(), 12345);
-		assert_eq!(val.at(1).unwrap().as_char().unwrap(), 'c');
-	}
-
-	#[test]
-	fn accessing_composites() {
-		// We already test accessing composite Values. This also checks that `at` works on
-		// the Composite type, too..
-
-		let val = Composite::named([("foo", Value::u128(12345)), ("bar", Value::char('c'))]);
-
-		assert_eq!(val.at("foo").unwrap().as_u128().unwrap(), 12345);
-		assert_eq!(val.at("bar").unwrap().as_char().unwrap(), 'c');
-
-		let val = Composite::unnamed([Value::u128(12345), Value::char('c')]);
-
-		assert_eq!(val.at(0).unwrap().as_u128().unwrap(), 12345);
-		assert_eq!(val.at(1).unwrap().as_char().unwrap(), 'c');
-	}
+    use super::*;
+
+    // This is basically the doc example with a little extra.
+    #[test]
+    fn nested_accessing() {
+        let val = Value::named_composite([(
+            "hello",
+            Value::unnamed_composite([
+                Value::u128(1),
+                Value::bool(true),
+                Value::named_composite([
+                    ("wibble", Value::bool(false)),
+                    ("foo", Value::named_composite([("bar", Value::u128(123))])),
+                ]),
+            ]),
+        )]);
+
+        assert_eq!(val.at("hello").at(0), Some(&Value::u128(1)));
+        assert_eq!(val.at("hello").at(1), Some(&Value::bool(true)));
+        assert_eq!(val.at("hello").at(2).at("wibble"), Some(&Value::bool(false)));
+        assert_eq!(val.at("hello").at(2).at("foo").at("bar"), Some(&Value::u128(123)));
+
+        assert_eq!(val.at("wibble").at(3), None);
+        assert_eq!(val.at("wibble").at("wobble").at("nope"), None);
+
+        // Strings can be used:
+        assert_eq!(val.at("hello").at(0), Some(&Value::u128(1)));
+        // References to valid locations are fine too:
+        assert_eq!(val.at("hello").at(0), Some(&Value::u128(1)));
```

Comment:
Nit: Should we remove this line?

Reply:
I updated the lines to actually do what the comments said :D